### ISSUE TITLE
Compatibility with other extensions like Title Window is Back

### ISSUE THREAD

**Issue by 0Raik** [neuromorph/openbar#24]
**Title Window is Back** makes a weird double shadow element.

**Just Perfection** is incompatible if using custom settings affecting the top bar like custom width and icons spacing. Setting everything to Theme Defaults makes everything happy.

**Workspace indicator by open apps** Show no island, its like its invisible. This might be a problem with how the developer made classes and calls and **Open Bar** probably can't do anything.

Lastly **Open Bar** setting to update colors when changing to a new wallpaper will reset some settings like borders which I set everything to hidden and 0 to have islands with background and nothing else. No shadow or border or glow but I have **Wallpaper Slideshow** and changes my wallpaper every X amount of time, when it does **Open Bar** resets my settings and I have to reopen and set back everything. I think this is a bug because some settings aren't changed. 

Thanks for the great work creating and maintaining **Open Bar** making Gnome super pretty and making theming quite obsolete. Have a wonderful day.

**Comment by neuromorph** [comment]
Hello,

Please let me know your Gnome and Extension versions so I can look into it.

**Title Window is Back**: I am not aware of it. I will look into it.

**Just Perfection**: When both extensions set the style settings, the later one will override previous one and the order of enabling on startup cannot be set. Though, I will look into it about the width and spacing.

**Workspace indicator by open apps**: I will try it out, need to install first. Gnome version pls.

**Open Bar**: Settings like border width, shadow getting reset was raised in a reddit post. So I have fixed it in the 'main' branch here on Github in 'autothemes.js' file. You can replace your file with it and that should fix it. It will appear in the next update.

I will get back on the above extensions when I get to try it out. Do report on versions.

> Thanks for the great work creating and maintaining Open Bar making Gnome super pretty and making theming quite obsolete. Have a wonderful day.

You are welcome and Thank you! Your comment is much appreciated!
Great day to you too!

**Comment by neuromorph** [comment]
**Window Title is Back**:
Looked into it. So they create a Panel Button which automatically gets style class 'panel-button'. However they then add another box into it and provide style class as 'panel-button' again. So now you have a Panel Button inside another Panel Button, creating the doubling effect. 
The easiest fix is to go into their `extension.js` and in the `class WindowTitleIndicator`, edit the following line to change the style class from 'panel-button' to something else (e.g. 'panel-button-box').
`this._box = new St.BoxLayout({style_class: 'panel-button'});`

**Workspace indicator by open apps**:
They create a regular widget (not a panel button) and add it to the panel in a non-standard way (without using 'main' API). This results in loosing some auto input handling and styles. Go to their `extension.js` and find following code (around line 201):  

`
// create indicator
    const indicator = new St.Bin({
      style: css_inline_workspace,
      style_class: css_classes_workspace.join(" "),
      reactive: true,
      can_focus: true,
      track_hover: true,
      child: new St.BoxLayout()
    })`

and replace it with:

`
// create indicator
    const indicator = new St.Bin({
      style: css_inline_workspace,
      style_class: css_classes_workspace.join(" "),
      reactive: true,
      can_focus: true,
      track_hover: true,
      child: new St.BoxLayout({reactive: true,
                              can_focus: true,
                              track_hover: true,
                              style_class: 'panel-button'})
    })`

Basically, when extensions do not use the standard API to add new buttons to the Panel, they lose style, menu, input handling etc. The above simple changes are quickest way to fix the issue without changing too much in their code.

**Comment by neuromorph** [comment]
**Just Perfection**:
I just tried setting panel size, indicator/button padding etc from Just Perfection settings while Open Bar already On. It did seem to work and changed the panel accordingly. However, as mentioned earlier, two extensions altering the styles of same widget can override each other and so I would not recommend it. Leave it at Theme Defaults in JP and set styles from Open Bar.

**Comment by 0Raik** [comment]
Amazing work!
You did mostly everything I was set to do which was to share your insights to the developers but you went far and beyond, thanks for contributing to the community :heart: 

Completely forgot to share my setup. Using Manjaro latest with Gnome 45. Took a while to reply because I'm offline living only going online every so often but use Gnome as a daily driver so thanks for the hard work.

Seems all issues have been addressed. I know **Just Perfection** settings are exclusive to use them or another extension own settings. I share it as a warning to others not to use both. If one sets **Just Perfection** to defaults and use **Open Bar**'s settings then everything works great. Solid work there.

Finally, thanks again and I will keep testing more as I truly fell in love with the extension and will surely share it everywhere. 

Have the best of luck and be safe, cheers!

**Comment by neuromorph** [comment]
Adding note for one more extension for documentation:

 **App Icons Taskbar:**
Issue: 
Taskbar buttons in the panel get cut at bottom.

Fix: 
Open the `stylesheet.css` file of this extension and make a tiny change from `3px` to `-3px` as below:
Existing code:
              `.azTaskbar-BaseIcon{
                 padding: 3px 8px;`
Change to:
            `.azTaskbar-BaseIcon{
                padding: -3px 8px;`
